File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = Infinity
    let best = 0
    
    for (let price of prices) {
        min = Math.min(min, price)
        best = Math.max(best, price - min)
    }
    return best
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        best = 0
        curr_min = prices[0]
        
        for p in prices:
            if p < curr_min:
                 curr_min = p
            curr = p - curr_min
            if best < curr:
                best = curr
                
        return best